demos/python_demos/multi_camera_multi_person_tracking/utils/network_wrappers.py

```diff
@@ -81,3 +105,215 @@ def forward(self, batch):
+
+    class Compose(object):
+        def __init__(self, transforms):
+            self.transforms = transforms
+
+        def __call__(self, sample):
+            for t in self.transforms:
+                sample = t(sample)
+            return sample
+
+    class Resize(object):
+        def __init__(self, max_size=None, window_size=None, size=None):
+            super().__init__()
+            assert int(max_size is not None) + int(window_size is not None) + int(size is not None) == 1
+
+            self.short_side_max = None
+            self.long_side_max = None
+            if max_size is not None:
+                self.short_side_max, self.long_side_max = max_size
+
+            self.height_max = None
+            self.width_max = None
+            if window_size is not None:
+                self.height_max, self.width_max = window_size
+
+            self.height = None
+            self.width = None
+            if size is not None:
+                self.height, self.width = size
+
+        def get_scale(self, image_size):
+            if self.height is not None:
+                scale_x, scale_y = self.width / image_size[1], self.height / image_size[0]
+            elif self.height_max is not None:
+                im_scale = min(self.height_max / image_size[0], self.width_max / image_size[1])
+                scale_x, scale_y = im_scale, im_scale
+            else:
+                im_scale = min(self.short_side_max / min(image_size), self.long_side_max / max(image_size))
+                scale_x, scale_y = im_scale, im_scale
+            return scale_x, scale_y
+
+        def __call__(self, sample):
+            image_size = sample['image'].shape[:2]
+            scale_x, scale_y = self.get_scale(image_size)
+
+            # Resize image.
+            sample['image'] = cv2.resize(sample['image'], None, fx=scale_x, fy=scale_y)
+            h, w = sample['image'].shape[:2]
+
+            # Resize boxes.
+            if 'gt_boxes' in sample:
+                sample['gt_boxes'] *= [scale_x, scale_y, scale_x, scale_y]
+                sample['gt_boxes'] = np.clip(sample['gt_boxes'], 0, [w - 1, h - 1, w - 1, h - 1])
+
+            # Resize masks.
+            if 'gt_masks' in sample:
+                sample['gt_masks'] = [[np.clip(part * [scale_x, scale_y], 0, [w - 1, h - 1]) for part in obj]
+                                      for obj in sample['gt_masks']]
+
+            return sample
+
+    class Normalize(object):
+        def __init__(self, mean, std):
+            self.mean = mean
+            self.std = std
+
+        def __call__(self, sample):
+            sample['image'] = sample['image'].astype('float32').transpose(2, 0, 1)
+            sample['image'][:, :, 0] = (sample['image'][:, :, 0] - self.mean[0]) / self.std[0]
+            sample['image'][:, :, 1] = (sample['image'][:, :, 1] - self.mean[1]) / self.std[1]
+            sample['image'][:, :, 2] = (sample['image'][:, :, 2] - self.mean[2]) / self.std[2]
```

Comment:
1. The channel dimension now stays first because of `transpose(2, 0, 1)`.
2. Use  broadcasting:
    `sample['image'] -= self.mean[:, None, None]`
    `sample['image'] /= self.std[:, None, None]`
    Or in one line but not in place: `sample['image'] = (sample['image'] - self.mean[:, None, None]) / self.std[:, None, None]`
    assuming you will replace `self.mean = mean` with `self.mean = np.array(mean)` in `__init__(self, mean, std)`.
3. But for me your version with wrong dimensions works better than my suggested version. So I question whether this normalization is needed. I think you should just drop it. Intel models usually have means and scales already fused into the model. `instance-segmentation-security-????` are not exceptions. (Assuming `Normalize` is only used for `instance-segmentation-security-????` models)

Reply:
`Normalize` removed.